nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/handler/LogRecordBuffer.java

```diff
@@ -70,25 +87,70 @@ public LogRecordBuffer(final int capacity) {
     public LogRecordBuffer(final int capacity, final int maxWait) {
         this.capacity = capacity;
         this.maxWait = maxWait;
-        this.pendingRecords = new ArrayBlockingQueue<>(capacity);
+        this.pendingRecords = new ArrayBlockingQueue<>(capacity, true);
+    }
+
+
+    /**
+     * Reconfigures the buffer.
+     *
+     * @param newCapacity capacity of the buffer.
+     * @param newMaxWait maximal time in seconds to wait for the free capacity. If &lt; 1, can wait
+     *            forever.
+     */
+    // R/W locks - only this method locks everything, because it has to recreate the queue using
+    // the original pending records.
+    public void reconfigure(final int newCapacity, final int newMaxWait) {
+        if (this.capacity == newCapacity && this.maxWait == newMaxWait) {
```

Comment:
This check is not thread-safe...

Reply:
I am thinking why I did that this way ... and maybe it is ok - those values can be changed just by this method in the locked part. Then if those values were changed by first thread, if the second brings same values it just leaves. If it has another, it will wait on the lock until the first finishes.
If the second thread would have original values, then it can return or can continue, BUT the same situation is with moving lock before the condition because the order of thread execution is not defined. The second can be processed first.

The only disadvantage would be if both first and second thread would have same values and both would pass the check, the first would do the reconfiguration, and the second would do that redundantly again.

But yeah, I can move it up, the method is not called frequently so there is no need to hurry with fast way out. The second thread can wait for the lock first; also the reconfiguration is very quick here.